### PR TITLE
Add Category and Contactperson to Indicator REST API for Spreadsheet

### PR DESCRIPTION
**Pull request by Bhavatu**
Related frontend: https://github.com/kausaltech/kausal-extensions/pull/43

### CODE REVIEW

File: indicators/api.py [hunk 1]
```diff
@@ -221,32 +226,252 @@ class Meta:
...
+            ct_cats = [cat.id for cat in cats if cat.type_id == ct.pk]
+            if ct.select_widget == ct.SelectWidget.SINGLE:
+                val = ct_cats[0] if len(ct_cats) else None
+            else:
+                val = ct_cats
+            out[ct.identifier] = val
+        return out
+
+
+
```
Comment (tituomin) [inline indicators/api.py]: Empty lines between methods a bit inconsistent (here 3 lines, elsewhere 2 or 1). I think 1 is our standard? Wonder why the linter doesn't complain.

File: indicators/api.py [hunk 2]
```diff
@@ -221,32 +226,252 @@ class Meta:
...
         )
 
     def create(self, validated_data: dict):
+        contact_persons_data = validated_data.pop('contact_persons', None)
+        categories_data = validated_data.pop('categories', None)
         instance = super().create(validated_data)
+
+        fields = cast(dict[str, serializers.Field], self.fields)
+
+        if categories_data is not None and hasattr(fields['categories'], 'update'):
```
Comment (tituomin) [inline indicators/api.py:444]: Just curious, why and when would the field (ie. serializer) not have the update method?

File: indicators/models.py [hunk 3]
```diff
@@ -600,7 +600,11 @@ def clean(self):
                                                  % self.common.unit)})
             # Unfortunately it seems we need to check whether dimensions are equal in the form
 
-    def set_categories(self, ctype: CategoryType, categories: list[int | Category]):
+    def set_categories(self, ctype: CategoryType, categories: list[int | Category], plan: Plan | None = None):
```
Comment (tituomin) [inline indicators/models.py]: CategoryType | str  ?

**Review by tituomin — APPROVED** [review]
Looks good, with some very minor comments.

File: indicators/api.py (hunk 2 above)
Comment (Bhavatu) [reply indicators/api.py:444]: Never really i can think of, but it made linter happy about attr-defined

**Pull request merged**